OrbitGl/App.cpp

```diff
@@ -870,56 +865,82 @@ void OrbitApp::SendErrorToUi(const std::string& title,
 }
 
 //-----------------------------------------------------------------------------
-void OrbitApp::EnqueueModuleToLoad(const std::shared_ptr<Module>& a_Module) {
-  m_ModulesToLoad.push_back(a_Module);
-}
+void OrbitApp::LoadModules(int32_t process_id,
+                           const std::vector<std::shared_ptr<Module>>& modules,
+                           const std::shared_ptr<Session>& session) {
+  // TODO(antonrohr) use ModuleData instead of Module
+  for (const auto& module : modules) {
+    {
+      absl::MutexLock lock(&modules_loading_mutex_);
+      if (modules_currently_loading_.contains(module->m_FullName)) {
+        continue;
+      }
+      modules_currently_loading_.insert(module->m_FullName);
+    }
+    thread_pool_->Schedule([this, module, process_id, session]() {
+      if (symbol_helper_.LoadSymbolsUsingSymbolsFile(module)) {
+        LOG("Loaded %lu function symbols locally for modules %s",
+            module->m_Pdb->GetFunctions().size(), module->m_Name);
+      } else {
+        LOG("Did not find local symbols for module: %s", module->m_Name);
+        const auto remote_symbols_result =
+            process_manager_->LoadSymbols(module->m_FullName);
+
+        if (!remote_symbols_result) {
+          SendErrorToUi(
+              "Error loading symbols",
+              absl::StrFormat(
+                  "Did not find symbols on local machine for module \"%s\". "
+                  "Trying to load symbols from remote resulted in error "
+                  "message: %s",
+                  module->m_Name, remote_symbols_result.error()));
+          {
+            absl::MutexLock lock(&modules_loading_mutex_);
+            modules_currently_loading_.erase(module->m_FullName);
+          }
+          return;
+        }
+        symbol_helper_.LoadSymbolsIntoModule(module,
+                                             remote_symbols_result.value());
+        LOG("Received and loaded %lu function symbols from remote service "
+            "for module %s",
+            module->m_Pdb->GetFunctions().size(), module->m_Name.c_str());
+      }
```

Comment:
maybe extract this to a separate method - then you can use return in if and drop else.

Reply:
this is a great idea. Makes the code way nicer. thank you! (done)